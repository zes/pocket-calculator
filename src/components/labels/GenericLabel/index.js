export { default } from './GenericLabel.component';
