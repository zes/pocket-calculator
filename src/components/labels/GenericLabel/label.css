.operation {
  color: #ffffff;
}

.result {
  color: #ffffff;
  font-weight: bold;
}