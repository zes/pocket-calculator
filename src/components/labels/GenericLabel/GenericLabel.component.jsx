import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import './label.css';

export default class GenericLabel extends PureComponent {
  render() {
    const { className, value } = this.props;

    return <span className={className}>{value}</span>;
  }
}

GenericLabel.propTypes = {
  className: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
};
