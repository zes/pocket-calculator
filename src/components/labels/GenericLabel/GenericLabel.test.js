import React from 'react';
import renderer from 'react-test-renderer';
import GenericLabel from './GenericLabel.component';

describe('[Component] <GenericLabel />', () => {
  const props = {
    className: 'operation',
    value: 'blablabla',
  };

  it('renders a snapshot', () => {
    const tree = renderer.create(<GenericLabel {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
