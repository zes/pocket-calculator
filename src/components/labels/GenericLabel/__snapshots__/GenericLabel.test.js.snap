// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component] <GenericLabel /> renders a snapshot 1`] = `
<span
  className="operation"
>
  blablabla
</span>
`;
